internal/hcp/registry/hcl_test.go

```diff
@@ -0,0 +1,157 @@
+package registry
+
+import (
+	"reflect"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/packer/hcl2template"
+)
+
+func TestRegisterAllBuilds(t *testing.T) {
+	cases := map[string]struct {
+		expectedBuilds       []string
+		err                  bool
+		diagsSummaryContains string
+		builds               *hcl2template.Builds
+	}{
+		"single build block with single source": {
+			expectedBuilds: []string{"docker.ubuntu"},
+			builds: &hcl2template.Builds{
+				&hcl2template.BuildBlock{
+					Sources: []hcl2template.SourceUseBlock{
+						{
+							SourceRef: hcl2template.SourceRef{
+								Type: "docker",
+								Name: "ubuntu",
+							},
+						},
+					},
+				},
+			},
+		},
+		"single build block with name and with single source": {
+			expectedBuilds: []string{"my-build-block.docker.ubuntu"},
+			builds: &hcl2template.Builds{
+				&hcl2template.BuildBlock{
+					Name: "my-build-block",
+					Sources: []hcl2template.SourceUseBlock{
+						{
+							SourceRef: hcl2template.SourceRef{
+								Type: "docker",
+								Name: "ubuntu",
+							},
+						},
+					},
+				},
+			},
+		},
+		"single build block with multiple sources": {
+			expectedBuilds: []string{"docker.alpine", "docker.ubuntu"},
+			builds: &hcl2template.Builds{
+				&hcl2template.BuildBlock{
+					Sources: []hcl2template.SourceUseBlock{
+						{
+							SourceRef: hcl2template.SourceRef{
+								Type: "docker",
+								Name: "ubuntu",
+							},
+						},
+						{
+							SourceRef: hcl2template.SourceRef{
+								Type: "docker",
+								Name: "alpine",
+							},
+						},
+					},
+				},
+			},
+		},
+		"single build block with name and multiple sources": {
```

Comment:
Just for clarity, I would add `err: false` to those test cases, I understand it's not strictly necessary as it's the zero-value, but it would make it clearer at first glance imho